include/ofi_util.h

```diff
@@ -504,6 +504,13 @@ OFI_DECLARE_CIRQUE(struct fi_cq_tagged_entry, util_comp_cirq);
 
 typedef void (*ofi_cq_progress_func)(struct util_cq *cq);
 
+struct util_cq_ctx {
+	struct util_comp_cirq	*cirq;
+	fi_addr_t		*src;
+	struct slist		aux_queue;
+	fi_cq_read_func		read_entry;
+};
```

Comment:
The cirq and src are already pointers that we don't need to allocate.  Having this structure reduces the size of the util_cq by 3 pointers total and introduces a significant amount of changes.  We don't expect apps to allocate more than a few, probably only 1, CQ.

This patch contains a plethora of updates to the providers.  Is there no way to break those out?

Reply:
I'll rework it to keep the resources in the util_cq and conditionally allocate it. That should reduce the changes for the providers but it will still touch all the providers to update the cq write calls. I'll see what I can do. Stand by.